AnonUpload/src/main/java/com/theironyard/controllers/AnonFileController.java

```diff
@@ -25,18 +25,54 @@
     AnonFileRepository repository;
 
     @RequestMapping(path = "/upload", method = RequestMethod.POST)
-    public void upload(MultipartFile file, HttpServletResponse response) throws IOException {
+    public void upload(MultipartFile file, String noDelete, String comment, String password, HttpServletResponse response) throws IOException, PasswordStorage.CannotPerformOperationException {
         File dir = new File("public/files");
         if(!dir.exists()) {
             dir.mkdirs();
         }
 
+        //Delete files, from database, if there are more than max allowed (max).
+        long max = 5;
+        while (repository.count() >= max) {
+            AnonFile deleteFile = repository.findFirstByPermanentFalseOrderByIdAsc();
+            repository.delete(deleteFile.getId());
+        }
+
         File f = File.createTempFile("anonfile", file.getOriginalFilename(), dir);
         FileOutputStream fos = new FileOutputStream(f);
         fos.write(file.getBytes());
 
         AnonFile anonFile = new AnonFile(f.getName(), file.getOriginalFilename());
+
+        //Did the user enter a comment?
+        if (!comment.isEmpty()){
```

Comment:
You could probably just set the comment.  If it is null then it doesn't hurt anything.